tests/components/history/conftest.py

```diff
@@ -3,15 +3,15 @@
 import pytest
 
 from homeassistant.components import history
+from homeassistant.components.recorder import Recorder
 from homeassistant.const import CONF_DOMAINS, CONF_ENTITIES, CONF_EXCLUDE, CONF_INCLUDE
-from homeassistant.setup import setup_component
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
 
 
 @pytest.fixture
-def hass_history(hass_recorder):
+async def hass_history(recorder_mock: Recorder, hass: HomeAssistant) -> None:
```

Comment:
```suggestion
async def hass_history(hass: HomeAssistant, recorder_mock: Recorder) -> None:
```

Reply:
This is not possible, the recorder fixture must be set up first

Reply:
The `hass` fixture is used by the `recorder_mock` fixture.

Reply:
The reason why we can't change it is explained here: https://github.com/home-assistant/core/pull/116448#discussion_r1588374159